src/java_bytecode/java_bytecode_parser.cpp

```diff
@@ -1509,3 +1652,65 @@ void java_bytecode_parsert::parse_local_variable_type_table(methodt &method)
       "Entry in LocalVariableTypeTable must be present in LVT");
   }
 }
+
+/// Read method handle pointed to from constant pool entry at index, return type
+/// of method handle and name if lambda function is found.
+/// \param entry: the constant pool entry of the methodhandle_info structure
+/// \returns: the method_handle type of the methodhandle_structure,
+/// either for a recognized bootstrap method or for a lambda function
+optionalt<java_bytecode_parsert::lambda_method_handlet>
+java_bytecode_parsert::parse_method_handle(const pool_entryt &entry)
+{
+  INVARIANT(
+    entry.tag == CONSTANT_MethodHandle,
+    "constant pool entry must be a MethodHandle");
+  lambda_method_handlet lambda_method_handle;
+  const auto &ref_entry = pool_entry(entry.ref2);
+  INVARIANT(
+    (entry.ref1 > 0 && entry.ref1 < 10),
+    "reference kind of Methodhandle must be in the range of 1 to 9");
+
+  const auto &class_entry = pool_entry(ref_entry.ref1);
+  const auto &nameandtype_entry = pool_entry(ref_entry.ref2);
+
```

Comment:
I think there should be a double check here that `name_and_type_entry.tag` is set to `CONSTANT_NameAndType`

Reply:
I believe this comment is addressed by mgudemann/cbmc#1 to see how I've done this.